src/alire/alire-hashes.ads

```diff
@@ -30,6 +30,31 @@ package Alire.Hashes with Preelaborate is
    function Digest (Hash : Any_Hash) return Any_Digest;
    --  Return the actual fingerprint without the kind prefix.
 
+   function Hash_Directory (Kind   : Kinds;
+                            Path   : Platform_Independent_Path;
+                            Except : Platform_Independent_Path := "")
+                            return Any_Hash;
+   --  Deterministic recursive hashing of a folder contents. The Except
+   --  folder, if given, is not included in the hashing nor recursed into. The
+   --  deterministic procedure followed is:
+   --  * All operations use utf8-encoded filenames.
+   --  * Enumerate current folder contents, and sort them alphabetically.
+   --    (Note that ordinary String comparison is used, so not utf8-order but
+   --    plain Character order is used.)
+   --    * ".", "..", Except entries are omitted from hashing.
+   --  * For each dir entry hash the following:
+   --    * UTF8-encoded name, with full route from "root" (Path).
```

Comment:
How is this handling Windows paths root?

Reply:
If I understand what you are asking, the fake root is our check out folder so it should not include anything platform-specific. E.g., if the checkout is of ```C:\Workspace\Crate```, then the hashing starts with ```/``` that maps to that folder.